users list implementation

Sam09040 opened the pull request:
## added:
- compare function moved to password.ts file;
- validation.ts file with a function to verify the token;
- end parameter to users query and  resolver;
- function to sort users array;
- default number of users showed, in case there's no end parameter;

Sam09040 committed:
users list implementation

## add:
- compare function to password.ts file;
- validation.ts file to verify the token;
- end parameter to users query and  resolver;
- sorting of users array;
- default number of users showed, in case there's no end parameter;

walteraandrade commented on src/app/graphql/resolvers.ts:
```diff
@@ -1,34 +1,53 @@
+      };
 
-      return users;
+      const sortUsers = (users: User[]): User[] => {
```
não é necessário fazer essa ordenação em javascript, [o prisma já oferece uma resposta ordenada
](https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#sorting)

walteraandrade commented on src/app/graphql/resolvers.ts:
```diff
@@ -1,34 +1,53 @@
+
+      let sortedUsers = sortUsers(users);
+
+      if (!end) {
```
[isso aqui pode simplificar um pouco essa lógica toda também](https://www.prisma.io/docs/orm/prisma-client/queries/pagination)

Sam09040 committed:
better logic on users query

## updated:
- using orderBy and take from prisma in order to sort and get certain number of users;

## removed:
- sortUsers function;

racsandoval commented on src/app/graphql/resolvers.ts:
```diff
@@ -1,25 +1,41 @@
+        end = 10;
+      }
+
+      const users = await prisma.user.findMany({
+        take: end,
+        orderBy: {
+          name: 'asc',
+        },
+      });
```
aqui conseguimos definir apenas o número de itens que o banco retorna (`take`) mas não conseguimos pular alguns itens (`offset`), como funciona a paginação sem utilizar o offset?

Sam09040 replied:
essa pr é só pra users query, a pr de paginação ainda vai ser feita.
